src/RemoteRequest/CurlRemoteGetRequest.php

```diff
@@ -132,7 +132,7 @@ static function ($curl, $header) use (&$headers) {
             $curlErrno = curl_errno($curlHandle);
             curl_close($curlHandle);
 
-            if ($body === false) {
+            if ($body === false || $status < 200 || $status >= 300) {
```

Comment:
This should work as expected, but note that I believe `$status` could be `0`. In this case, `$body` should also be `false`.

Reply:
Hey! I am not sure, if I understand you correctly. Is there anything to do left?

If `$status` can be `0`, it should IMHO be treated as error. And this code does that (smaller than `200`).

Reply:
Yeah, I'm saying the same thing. In the error case, `$status` should be `0` and it will work as expected.